### PR TITLE
Convert ReaderAvatar to TypeScript

### PR DESCRIPTION
**Pull request by mashikag**
Related to https://github.com/Automattic/wp-calypso/issues/77792

## Proposed Changes

* Convert ReaderAvatar to TypeScript

## Testing Instructions

* Ensure that there is no typing issues introduced, and that the bundle builds fine
* Ensure that there is no visible regressions in reader, wherever the `ReaderAvatar` may be used:
http://calypso.localhost:3000/read 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/blocks/reader-avatar/index.tsx [hunk 1]
```diff
@@ -20,18 +38,16 @@ const ReaderAvatar = ( {
...
-	// don't show the default favicon for some sites
-	if ( feedIcon?.endsWith( 'wp.com/i/buttonw-com.png' ) ) {
-		feedIcon = null;
-	}
-
 	const safeSiteIcon = safeImageUrl( siteIcon );
-	const safeFeedIcon = safeImageUrl( feedIcon );
+	const safeFeedIcon = safeImageUrl(
+		// don't show the default favicon for some sites
+		feedIcon?.endsWith( 'wp.com/i/buttonw-com.png' ) ? undefined : feedIcon
```
Comment (TimBroddin) [inline client/blocks/reader-avatar/index.tsx]: Just semantics, but why not null?

**Review by TimBroddin — CHANGES_REQUESTED** [review]
I did `yarn build` & `yarn build-packages`, all good.

I did however got this error in my console when visiting my WP.com dashboard: 

CleanShot 2023-06-07 at 15 30 34@2x
 
This is caused by the removal of `dispatch` from the function definition of MediaFile.

**Comment by mashikag** [comment]
> I did yarn build & yarn build-packages, all good.
> 
> I did however got this error in my console when visiting my WP.com dashboard

You mean when you tried to open: `http://calypso.localhost:3000/home/<site-slug>`?

File: client/blocks/reader-avatar/index.tsx (hunk 1 above)
Comment (mashikag) [reply client/blocks/reader-avatar/index.tsx]: I did not want to alter the original type definition of the `safeImageUrl` function. According to the definition, the param could be only either `string | undefined`. However, I examined the function now, and I think it makes sense to allow `null` too, to account for such use case. Done 👍

**Comment by mashikag** [comment]
> > I did yarn build & yarn build-packages, all good.
> > I did however got this error in my console when visiting my WP.com dashboard
> 
> You mean when you tried to open: `http://calypso.localhost:3000/home/<site-slug>`?

For future reference, the conversation moved to: p1686146162135479/1686138287.968289-slack-C02TCEHP3HA

**Review by TimBroddin — APPROVED** [review]
No more error messages!

**Pull request merged**